src/utils/helper.ts

```diff
@@ -209,7 +209,18 @@ export const getDestinationAddress = (swap: SomeSwap) => {
     }
 
     if (swap.type === SwapType.Submarine) {
-        return (swap as SubmarineSwap).invoice;
+        const submarineSwap = swap as SubmarineSwap;
+        return submarineSwap.originalDestination || submarineSwap.invoice;
+    }
+
+    if (swap.type === SwapType.Chain) {
+        const chainSwap = swap as ChainSwap;
+        return chainSwap.originalDestination || chainSwap.claimAddress;
+    }
+
+    if (swap.type === SwapType.Reverse) {
+        const reverseSwap = swap as ReverseSwap;
+        return reverseSwap.originalDestination || reverseSwap.claimAddress;
```

Comment:
These two have an overlap

```suggestion
    if (swap.type === SwapType.Reverse || SwapType.Chain) {
        const chainSwap = swap as ReverseSwap | ChainSwap ;
        return chainSwap.originalDestination || chainSwap.claimAddress;
```